### PR TITLE
feat(radar_fusion_to_detected_object): add yaw gate

### PR DESCRIPTION
**Pull request by scepter914**
Signed-off-by: scepter914 <scepter914@gmail.com>

## Description

Add yaw gate in radar_fusion_to_detected_object.

## Tests performed

Test by rosbag.

- Before

Screenshot from 2022-07-07 17-17-26

- After
Screenshot from 2022-07-07 17-28-44

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
I think it is better to use [object_merger](https://github.com/autowarefoundation/autoware.universe/tree/main/perception/object_merger) algorithm to fuse radar object and lidar object. How about you?

File: perception/radar_fusion_to_detected_object/src/radar_fusion_to_detected_object.cpp [hunk 1]
```diff
@@ -97,12 +98,20 @@ RadarFusionToDetectedObject::Output RadarFusionToDetectedObject::update(
...
       if (!radars_within_split_object || !(*radars_within_split_object).empty()) {
-        split_object.kinematics.has_twist = true;
-        split_object.kinematics.twist_with_covariance =
+        TwistWithCovariance twist_with_covariance =
           estimateTwist(split_object, radars_within_split_object);
+        double twist_yaw = tier4_autoware_utils::normalizeRadian(
+          std::atan2(twist_with_covariance.twist.linear.y, twist_with_covariance.twist.linear.x));
+        double object_yaw = tier4_autoware_utils::normalizeRadian(
+          tf2::getYaw(split_object.kinematics.pose_with_covariance.pose.orientation));
+        ;
```
Comment (yukkysaito) [inline perception/radar_fusion_to_detected_object/src/radar_fusion_to_detected_object.cpp:109]: typo?
```suggestion
```

File: perception/radar_fusion_to_detected_object/src/radar_fusion_to_detected_object.cpp [hunk 2]
```diff
@@ -97,12 +98,20 @@ RadarFusionToDetectedObject::Output RadarFusionToDetectedObject::update(
...
+        double diff_yaw = tier4_autoware_utils::normalizeRadian(twist_yaw - object_yaw);
```
Comment (yukkysaito) [inline perception/radar_fusion_to_detected_object/src/radar_fusion_to_detected_object.cpp:110]:
```suggestion
        const TwistWithCovariance twist_with_covariance =
          estimateTwist(split_object, radars_within_split_object);
        const double twist_yaw = tier4_autoware_utils::normalizeRadian(
          std::atan2(twist_with_covariance.twist.linear.y, twist_with_covariance.twist.linear.x));
        const double object_yaw = tier4_autoware_utils::normalizeRadian(
          tf2::getYaw(split_object.kinematics.pose_with_covariance.pose.orientation));
        ;
        const double diff_yaw = tier4_autoware_utils::normalizeRadian(twist_yaw - object_yaw);
```

**Comment by yukkysaito** [comment]
Almost Ok :+1:

**Comment by scepter914** [comment]
> I think it is better to use [object_merger](https://github.com/autowarefoundation/autoware.universe/tree/main/perception/object_merger) algorithm to fuse radar object and lidar object. How about you?

Conclusion from discussing
- 1:1 object merge (Example: `object_merger`, `multi_object_tracking`)
Use `object_merger`, and will make util library for object merger

- 1: n object merge
Example: `radar_fusion_to_detected_object`, sensor fusion tracker
Write object merge algorithm for each package

File: perception/radar_fusion_to_detected_object/src/radar_fusion_to_detected_object.cpp (hunk 2 above)
Comment (scepter914) [reply perception/radar_fusion_to_detected_object/src/radar_fusion_to_detected_object.cpp]: fixed

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**